### PR TITLE
shell: unset 'active' state also when closing the menu by clicking on the same iframe

### LINKED ISSUE #18182: mobile: When in responsive mode, the system navigation menu pops up again when unasked for
Steps to reproduce:

1. Resize browser to trigger mobile breakpoint. (You don't have to use a mobile browser or even mobile emulation.)
2. Switch pages to something that's not cluttered (so you're not clicking on something accidentally), such as SELinux or Diagnostic Reports.  This happens all the time on the terminal page as well.
3. Click (or tap) on the background of the page.

My guess is that it's either related to focus (possibly with bubbling too) activating the menu again. And/or it's using some kind of a `toggle()` without explicitly setting to true/false.

This breaks mobile functionality everywhere. It can also break desktop browsers too, but only when resized to be small and/or using large enough fonts to trigger the issue. (Both of these things can happen when used "in the wild".)

_Originally posted by @garrett in https://github.com/cockpit-project/cockpit/issues/18157#issuecomment-1384277486_

### PR DESCRIPTION
**Pull request by KKoukiou**
See https://github.com/cockpit-project/cockpit/issues/18182#issuecomment-1401641783

Fixes #18182

### CODE REVIEW

**Review by garrett — CHANGES_REQUESTED** [review]
Looks great, except for the missing semicolon. Fix that and I approve. Thanks for the comment in the code too!

**Comment by garrett** [comment]
Oh, I tested in Chrome & Firefox on Android, Safari on iOS. Seemed to work as expected! Thanks again!

**Review by martinpitt — APPROVED** [review]
Thanks @garrett ! I applied your suggestion, will squash it during landing.

**Review by garrett — APPROVED** [review]
With the semicolon, this gets an approval! Thanks again! :tada:

File: pkg/shell/nav.jsx [hunk 1]
```diff
@@ -17,11 +17,23 @@ export const SidebarToggle = () => {
...
+            setActive(false);
+        };
 
-        window.addEventListener("blur", handleClickOutside);
+        ["blur", "click"].map(ev_type => window.addEventListener(ev_type, handleClickOutside));
 
-        return () => window.removeEventListener("blur", handleClickOutside);
+        return () => {
+            ["blur", "click"].map(ev_type => window.removeEventListener(ev_type, handleClickOutside));
+        };
```
Comment (cockpituous) [inline pkg/shell/nav.jsx:36]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18232-20230124-161236-853896e9-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**